### ISSUE TITLE
Checking Swap and Force times in the Algorithm 

### ISSUE THREAD

**Issue by MuellMark** [MuellMark/Course-Scheduler#240]
I've talked with colby, and we have an idea for how to add in swap and force courses. However, I need to make sure my code will be prepared for it and I need to make sure it exports correctly. I should do #239 first to make sure I can compare exports, but after that this should be worked on.

**Comment by MuellMark** [comment]
Colby will be adding in an extra <forced> tag on the csv's. With how it's written, it should work, but I need to double check and make sure that those tags don;t stay there

Same goes for the swap method, but forcing courses will be first

**Comment by MuellMark** [comment]
I can confirm that having two forced blocks is just consolidated into one, which is perfect. 

 Now onto swapping courses

**Comment by colbySullivan** [comment]
I am sitting here with Mark and we're trying to connect the two sides.

**Comment by colbySullivan** [comment]
We have cleanup some format issues but we are still working on getting the forced section to work.

**Comment by colbySullivan** [comment]
Mark and I have gotten the forced to work. We still need to clean up some sections but I am comfortable that we can get it done.

**Comment by colbySullivan** [comment]
I have gotten the flask API calls to correctly work and it starts on the home page rather than the CSV creation page now. I am going work on getting the swapped to work.

**Comment by colbySullivan** [comment]
I took a small break but I bounced some ideas off of Mark to see how best to implement the swapped functionality. I am going to continue working on that.

**Comment by colbySullivan** [comment]
I got a little carried away but I have been working on fixing the final output. I think I am going to sign off for tonight.

**Comment by MuellMark** [comment]
Colby sent me some CSV files that weren't working for swapping so I'm going to go ahead and see if I can figure out what's wrong, and fix it if it's on my end

**Comment by MuellMark** [comment]
It's giving me that they're infeasible. I need to check if they actually are infeasible, and if they are they should behave differently, maybe. I'm not sure if I need to overwrite certain files or not...

After looking into it, I can confirm that yes it is infeasible (as cs2 can't be on a tuesday/thursday time like it was trying to be). I'm going to go ahead and make some new test files that are very basic but will let us easily test swapping from the web page. 

I made a new CSV file for us to use that is very unrealistic but should also allow any courses to be swapped with anything, just for testing. That being said, I think there is something wrong with swapping since it's still very finicky. I think I forgot to add in code to cover if the forced tag is empty so I'll try and fix that. 

On top of that, I also think that there is some sort of issue with when a file is infeasible, but I'm not entirely sure what it is, so I'll be looking into both of these in the next half hour.

**Comment by MuellMark** [comment]
The issue with the forced column was actually just an issue with a conditional that I misplaced. That is now fixed. However, something else if definitely wrong. I'm trying to swap cs2 and cap but it reassigned them to seemingly random times:
<img width="243" alt="Screenshot 2024-04-22 at 11 53 56 PM" src="https://github.com/MuellMark/Course-Scheduler/assets/92736704/e7898a7d-79ea-4d9f-bfed-055df43e7a6e">
Looking into this now...

I can confirm the file is correct, I'm just not sure where the different times are coming from. One thing I am noticing is that because the forced courses are added at the top I should get rid of this while loop as it is no useless and needlessly confusing. Removing this didn;t fix anything though...

I'm going in and addin a lot of print statements to try and find whats wrong. I think I finally figured it out. There are two output.csv files and that must be what's throwing it off. I moved a file around and it seems to be working now, I;m going to try and do some quick tests.

Everything seems to be working, but I definitley need to get with colby and make sure I didn;t break something moving the file. I still want to do a few more tests later tomorrow to verify everything. I'm a little ticked off that it took so long to figure it out, but I did get to remove a useless part of my code, so I'm taking this as a win

**Comment by MuellMark** [comment]
I'm meeting with colby later today so I'm going to do a little testing and finish the clean up before then

**Comment by MuellMark** [comment]
I added better comments to the swap for future use, as the ones already in place were not super helpful. I've been doing tests and I think it's finally working as expected. I want to do a few more tests with infeasibility but I think it's in good shape for later today when I meet with colby

**Comment by MuellMark** [comment]
Colby is here so we are going to work together to get everything in working order to record for tomorrow

**Comment by MuellMark** [comment]
I merged my code and we got to brain storming on what different features are feasible for the presentation. I believe we'll have swapping and forcing times working, just not looking very pretty. We may even do a live demo of it. 

I need to change how the infeasibility is reported, as it is deleting everything in the file, which it really shouldn;t be doing. I'll have more details in a seperate issue, but for now I'll turn my attention to the presentation
